src/driver/firebird/FirebirdDriver.ts

```diff
@@ -0,0 +1,387 @@
+     * Firebrid database (no pooling)
+     */
+    firebirdDatabase: Database;
+
+    constructor(connection: Connection) {
+        this.connection = connection;
+        this.options = connection.options as FirebirdConnectionOptions;
+        this.firebirdOptions = connection.options as Options;
+
+        // load mysql package
+        this.firebird = PlatformTools.load("node-firebird");
+    }
+
+    async connect(): Promise<void> {
+        return new Promise<void>((ok, fail) => {
+            if (this.options.pooling) {
+                this.firebirdPool = this.firebird.pool(5, this.firebirdOptions);
+                this.firebirdPool.get((err, database) => {
+                    if (err) {
+                        fail(err);
+                    }
+                    this.firebirdDatabase = database;
+                    ok();
+                });
+            } else {
+                this.firebird.attachOrCreate(this.firebirdOptions, (err: any, database: Database) => {
+                    if (err) {
+                        fail(err);
+                    }
+                    this.firebirdDatabase = database;
+                    ok();
+                });
+            }
+        });
+    }
+
+    afterConnect(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    async disconnect(): Promise<void> {
+        return new Promise<void>((ok, fail) => {
+            if (this.options.pooling) {
+                this.firebirdPool.destroy();
+                ok();
+            } else {
+                this.firebirdDatabase.detach(() => ok());
+            }
+        });
+    }
+
+    createSchemaBuilder(): SchemaBuilder {
+        return new RdbmsSchemaBuilder(this.connection);
+    }
+    createQueryRunner(mode: "master" | "slave"): QueryRunner {
+        return new FirebirdQueryRunner(this);
+    }
+    escapeQueryWithParameters(sql: string, parameters: ObjectLiteral, nativeParameters: ObjectLiteral): [string, any[]] {
+        const escapedParameters: any[] = Object.keys(nativeParameters).map(key => nativeParameters[key]);
+        if (!parameters || !Object.keys(parameters).length)
+            return [sql, escapedParameters];
+
+        const keys = Object.keys(parameters).map(parameter => "(:(\\.\\.\\.)?" + parameter + "\\b)").join("|");
+        sql = sql.replace(new RegExp(keys, "g"), (key: string) => {
+            let value: any;
+            if (key.substr(0, 4) === ":...") {
+                value = parameters[key.substr(4)];
+            } else {
+                value = parameters[key.substr(1)];
+            }
+
+            if (value instanceof Function) {
+                return value();
+
+            } else {
+                escapedParameters.push(value);
+                return "?";
+            }
+        }); // todo: make replace only in value statements, otherwise problems
+        return [sql, escapedParameters];
+    }
+    escape(columnName: string): string {
+        return "`" + columnName + "`";
```

Comment:
```suggestion
        return '"' + columnName + '"';
```

Backticks does not work in firebird (we're using V3.0)